### PR TITLE
Fix of creating application in VSUP registrar module

### PR DESCRIPTION
**Pull request by metodej**
- merging of new application with user already in perun based on birth number in VSUP module now has to be approved manually by vo admin
- at the end of approval of application the user external source is created based on RC for future merging of the user

### CODE REVIEW

File: perun-registrar-lib/src/main/java/cz/metacentrum/perun/registrar/modules/Vsup.java
Comment (zlamalp) [inline perun-registrar-lib/src/main/java/cz/metacentrum/perun/registrar/modules/Vsup.java:210]: This would create ExtSource with name RC and type as value of birthNumber. At VŠUP type of RC ExtSource is ``INTERNAL``, we should have constant for it somewhere.

File: perun-registrar-lib/src/main/java/cz/metacentrum/perun/registrar/modules/Vsup.java
Comment (zlamalp) [inline perun-registrar-lib/src/main/java/cz/metacentrum/perun/registrar/modules/Vsup.java:211]: You shouldn't pass identity from application into the RC UserExtSource. It should contain login from the rc, like 
```java
new UserExtSource(es, 0, rc));
```

File: perun-registrar-lib/src/main/java/cz/metacentrum/perun/registrar/modules/Vsup.java
Comment (zlamalp) [inline perun-registrar-lib/src/main/java/cz/metacentrum/perun/registrar/modules/Vsup.java:62]: Message will contain toString() from the User object, which is not most user friendly way. I would prefer printing just displayName and ID.

**Review by zlamalp — CHANGES_REQUESTED** [review]

File: perun-registrar-lib/src/main/java/cz/metacentrum/perun/registrar/modules/Vsup.java
Comment (metodej) [reply perun-registrar-lib/src/main/java/cz/metacentrum/perun/registrar/modules/Vsup.java]: Fixed.

File: perun-registrar-lib/src/main/java/cz/metacentrum/perun/registrar/modules/Vsup.java
Comment (metodej) [reply perun-registrar-lib/src/main/java/cz/metacentrum/perun/registrar/modules/Vsup.java]: Fixed.

File: perun-registrar-lib/src/main/java/cz/metacentrum/perun/registrar/modules/Vsup.java
Comment (metodej) [reply perun-registrar-lib/src/main/java/cz/metacentrum/perun/registrar/modules/Vsup.java]: Replaced with `INTERVAL` constant.

File: perun-registrar-lib/src/main/java/cz/metacentrum/perun/registrar/modules/Vsup.java
Comment (stavamichal) [inline perun-registrar-lib/src/main/java/cz/metacentrum/perun/registrar/modules/Vsup.java:68]: Are you sure this should react to any exception? Because if there will be for example "InternalErrorException", this will say that the application can be approved if I understand it right.
@zlamalp should it work this way?

File: perun-registrar-lib/src/main/java/cz/metacentrum/perun/registrar/modules/Vsup.java
Comment (zlamalp) [inline perun-registrar-lib/src/main/java/cz/metacentrum/perun/registrar/modules/Vsup.java:102]: I realized, that this won't handle the cases, when found user is already a member, since after this method, normal approval process takes place. So here, for case when user is not null, please add check, whether user is already a member of VO, eg.:
```java
try {
	Member member = ((PerunBl) session.getPerun()).getMembersManagerBl().getMemberByUser(session, app.getVo(), app.getUser());
	// user is already a member, switch application type
	if (Application.AppType.INITIAL.equals(app.getType())) {
		app.setType(Application.AppType.EXTENSION);
		registrar.updateApplicationType(session, app);
		log.debug("Updating application type to EXTENSION since we matched user which is VO member!");
	}
} catch (MemberNotExistsException e) {
	// OK state
}
```
You must also implement new method in the RegistrarManager to update application type in a DB. It must be public, but there is no need to add it to the RPC, so we don't need authorization in it yet.

**Review by zlamalp — CHANGES_REQUESTED** [review]

File: perun-registrar-lib/src/main/java/cz/metacentrum/perun/registrar/modules/Vsup.java
Comment (metodej) [reply perun-registrar-lib/src/main/java/cz/metacentrum/perun/registrar/modules/Vsup.java:68]: Ok, I've change the catched exception only to those that are thrown by `getUserByExtSourceNameAndExtLogin()`.

File: perun-registrar-lib/src/main/java/cz/metacentrum/perun/registrar/modules/Vsup.java
Comment (metodej) [reply perun-registrar-lib/src/main/java/cz/metacentrum/perun/registrar/modules/Vsup.java]: Fixed.

File: perun-registrar-lib/src/main/java/cz/metacentrum/perun/registrar/modules/Vsup.java
Comment (metodej) [reply perun-registrar-lib/src/main/java/cz/metacentrum/perun/registrar/modules/Vsup.java:102]: Okay, thank you for comment, I've added this check and also the mentioned method.

**Review by stavamichal — APPROVED** [review]

File: perun-registrar-lib/src/main/java/cz/metacentrum/perun/registrar/modules/Vsup.java
Comment (Vojtech-Sassmann) [inline perun-registrar-lib/src/main/java/cz/metacentrum/perun/registrar/modules/Vsup.java:115]: Isn't here the `app.getUser()` always null? This should probably be located elsewhere.

**Review by Vojtech-Sassmann — CHANGES_REQUESTED** [review]

File: perun-registrar-lib/src/main/java/cz/metacentrum/perun/registrar/modules/Vsup.java
Comment (metodej) [reply perun-registrar-lib/src/main/java/cz/metacentrum/perun/registrar/modules/Vsup.java:114]: Yes, it should have been in the other branch of if, thank you.

**Review by Vojtech-Sassmann — APPROVED** [review]

File: perun-registrar-lib/src/main/java/cz/metacentrum/perun/registrar/impl/RegistrarManagerImpl.java
Comment (zlamalp) [inline perun-registrar-lib/src/main/java/cz/metacentrum/perun/registrar/impl/RegistrarManagerImpl.java:3651]: Must be ``application.getType().toString()`` otherwise jdbc driver cannot guess the type: ```Can't infer the SQL type to use for an instance of cz.metacentrum.perun.registrar.model.Application$AppType.```

**Review by zlamalp — CHANGES_REQUESTED** [review]

File: perun-registrar-lib/src/main/java/cz/metacentrum/perun/registrar/impl/RegistrarManagerImpl.java
Comment (metodej) [reply perun-registrar-lib/src/main/java/cz/metacentrum/perun/registrar/impl/RegistrarManagerImpl.java]: Ok, fixed.

**Review by zlamalp — APPROVED** [review]

**Pull request merged**
